tests/PHPStan/Analyser/nsrt/bug-7387.php

```diff
@@ -8,60 +8,61 @@ class HelloWorld
-		assertType('numeric-string', sprintf('%14g', $i));
-		assertType('numeric-string', sprintf('%14G', $i));
-		assertType('numeric-string', sprintf('%14h', $i));
-		assertType('numeric-string', sprintf('%14H', $i));
-		assertType('numeric-string', sprintf('%14o', $i));
-		assertType('numeric-string', sprintf('%14u', $i));
-		assertType('numeric-string', sprintf('%14x', $i));
-		assertType('numeric-string', sprintf('%14X', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14d', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14e', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14E', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14f', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14F', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14g', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14G', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14h', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14H', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14o', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14u', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14x', $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%14X', $i));
 
 	}
 
 	public function positionalArgs($mixed, int $i, float $f, string $s) {
 		// https://3v4l.org/vVL0c
 		assertType('non-falsy-string', sprintf('%2$14s', $mixed, $i));
 
-		assertType('numeric-string', sprintf('%2$.14F', $mixed, $i));
-		assertType('numeric-string', sprintf('%2$.14F', $mixed, $f));
+		assertType('non-empty-string&numeric-string', sprintf('%2$.14F', $mixed, $i));
+		assertType('non-empty-string&numeric-string', sprintf('%2$.14F', $mixed, $f));
 		assertType('numeric-string', sprintf('%2$.14F', $mixed, $s));
 
-		assertType('numeric-string', sprintf('%2$1.14F', $mixed, $i));
-		assertType('numeric-string', sprintf('%2$2.14F', $mixed, $f));
-		assertType('numeric-string', sprintf('%2$3.14F', $mixed, $s));
+		assertType('non-empty-string&numeric-string', sprintf('%2$1.14F', $mixed, $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%2$2.14F', $mixed, $f));
+		assertType('non-falsy-string&numeric-string', sprintf('%2$3.14F', $mixed, $s));
 
-		assertType('numeric-string', sprintf('%2$14F', $mixed, $i));
-		assertType('numeric-string', sprintf('%2$14F', $mixed, $f));
-		assertType('numeric-string', sprintf('%2$14F', $mixed, $s));
+		assertType('non-falsy-string&numeric-string', sprintf('%2$14F', $mixed, $i));
+		assertType('non-falsy-string&numeric-string', sprintf('%2$14F', $mixed, $f));
+		assertType('non-falsy-string&numeric-string', sprintf('%2$14F', $mixed, $s));
 
-		assertType('string', sprintf('%10$14F', $mixed, $s));
+		// XXX should be string because of invalid arguments count
+		assertType('non-falsy-string&numeric-string', sprintf('%10$14F', $mixed, $s));
```

Comment:
after https://github.com/phpstan/phpstan-src/pull/3126 is merged we have the necessary `PrintfHelper` to decide about the invalid number of arguments/placeholders